### DISCUSSION TITLE
`NotSame` should fail if args are not pointers

### DISCUSSION THREAD

**Discussion by n4nn31355** [stretchr/testify#1660] [Ideas]
It's confusing and error-prone that `NotSame` checks nothing if any of the values is not a pointer.
Is there a reason for this, or may we consider it a bug that needs an issue?
```
a := struct{ byte }{}
require.NotSame(t, a, a)   // Doesn't fail
require.NotSame(t, &a, &a) // Fails
```

**Comment by brackendawson** [comment]
It's a bug, I've opened #1661
